src/transaction.ts

```diff
@@ -579,26 +726,39 @@ class Transaction extends DatastoreRequest {
   /**
    * This function parses results from a beginTransaction call
    *
-   * @param {RequestPromiseReturnType} response The response from a call to
-   * begin a transaction.
-   * @param {RunCallback} callback A callback that accepts an error and a
+   * @param {UserCallbackData<google.datastore.v1.IBeginTransactionResponse>} [response]
+   * The response data from a call to begin a transaction.
+   * @param {RunCallback} [callback] A callback that accepts an error and a
    * response as arguments.
    *
    **/
   #processBeginResults(
-    response: RequestPromiseReturnType,
+    response: UserCallbackData<google.datastore.v1.IBeginTransactionResponse>,
     callback: RunCallback
   ): void {
     const err = response.err;
     const resp = response.resp;
     if (err) {
       callback(err, null, resp);
     } else {
-      this.id = resp!.transaction;
+      this.#parseRunSuccess(response);
       callback(null, this, resp);
     }
   }
 
+  /**
+   * This function saves results from a successful beginTransaction call.
+   *
+   * @param {UserCallbackData<any>} [response] The response from a call to
+   * begin a transaction that completed successfully.
+   *
+   **/
+  #parseRunSuccess(response: UserCallbackData<any>) {
+    const resp = response.resp;
+    this.id = resp!.transaction;
+    this.#state = TransactionState.IN_PROGRESS;
```

Comment:
No. Unfortunately the id in code has a dual purpose. Normally id stores the transaction id returned from beginTransaction. It can also be used to specify the retry transaction id in the first `beginTransaction` call by supplying id in the constructor and saving it at https://github.com/googleapis/nodejs-datastore/blob/75949b51ea37f2e316a0818b969796a98ce2ba10/src/transaction.ts#L77. If the user supplies the id in the constructor then we still want `state === NOT_STARTED` so we can't rely on `id` to set `state` to `NOT_STARTED` or `IN_PROGRESS`.